great_expectations/dataset/sqlalchemy_dataset.py

```diff
@@ -614,46 +632,138 @@ def get_column_median(self, column):
+            for quantile in quantiles
+        ]
+        quantiles_query: Select = sa.select(selects).select_from(self._table)
+
+        try:
+            quantiles_results: RowProxy = self.engine.execute(
+                quantiles_query
+            ).fetchone()
+            return list(quantiles_results)
+        except ProgrammingError as pe:
+            exception_message: str = "An SQL syntax Exception occurred."
+            exception_traceback: str = traceback.format_exc()
+            exception_message += f'{type(pe).__name__}: "{str(pe)}".  Traceback: "{exception_traceback}".'
+            logger.error(exception_message)
+            raise pe
+
+    def _get_column_quantiles_mysql(self, column: str, quantiles: tuple) -> list:
+        # MySQL does not support "percentile_disc", so we implement it as a compound query.
+        # Please see https://stackoverflow.com/questions/19770026/calculate-percentile-value-using-mysql for reference.
+        percent_rank_query: CTE = sa.select(
+            [
+                sa.column(column),
+                sa.func.percent_rank()
+                .over(order_by=sa.column(column).desc())
+                .label("p"),
             ]
+        ).order_by(sa.column("p").desc()).select_from(self._table).cte("t")
+
+        selects: List[WithinGroup] = []
+        for idx, quantile in enumerate(quantiles):
+            quantile_column: Label = sa.func.first_value(sa.column(column)).over(
+                order_by=sa.case(
+                    [(percent_rank_query.c.p < quantile, percent_rank_query.c.p)],
+                    else_=None,
+                ).desc()
+            ).label(f"q_{idx}")
+            selects.append(quantile_column)
+        quantiles_query: Select = sa.select(selects).distinct().order_by(
+            percent_rank_query.c.p.desc()
+        )
 
         try:
-            quantiles = self.engine.execute(
-                sa.select(selects).select_from(self._table)
+            quantiles_results: RowProxy = self.engine.execute(
+                quantiles_query
             ).fetchone()
+            return list(quantiles_results)
+        except ProgrammingError as pe:
+            exception_message: str = "An SQL syntax Exception occurred."
+            exception_traceback: str = traceback.format_exc()
+            exception_message += f'{type(pe).__name__}: "{str(pe)}".  Traceback: "{exception_traceback}".'
+            logger.error(exception_message)
+            raise pe
+
+    # Support for computing the quantiles column for PostGreSQL and Redshift is included in the same method as that for
+    # the generic sqlalchemy compatible DBMS engine, because distinguishing between them is oftentimes impossible.  The
+    # only difference is that Redshift and/or PostGreSQL sometimes do not support the aggregate function
+    # "percentile_disc", but do support the approximate percentile_disc or percentile_cont function version instead.
```

Comment:
```suggestion
    # Support for computing the quantiles column for PostGreSQL and Redshift is included in the same method as that for
    # the generic sqlalchemy compatible DBMS engine, because users often use the postgresql driver to connect to Redshift
    # The key functional difference is that Redshift does not support the aggregate function
    # "percentile_disc", but does support the approximate percentile_disc or percentile_cont function version instead.```